### ISSUE TITLE
Add YAML support

### ISSUE THREAD

**Issue by wmertens** [atom/first-mate#36]
**TL;DR**: With YAML you can write `begin: \bstart\b` instead of `begin: "\\bstart\\b"`.

I know YAML and CSON are quite similar, but YAML allows strings to remain unquoted (http://www.yaml.org/spec/1.2/spec.html#id2788859), which allows writing bare regexes and makes writing things like https://github.com/wmertens/sublime-nix/blob/master/nix.YAML-tmLanguage a lot easier.

It is also what is used by many grammars for Sublime, via the AAAPackageDev tools, and github allows it in linguist.

[Here's a YAML-tmLanguage highlighter used on github to highlight a YAML-tmLanguage file](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fgithub.com%2FSublimeText%2FAAAPackageDev%2Fblob%2Fmaster%2FSyntax%2520Definitions%2FSublime%2520Text%2520Syntax%2520Def%2520%28YAML%29.YAML-tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fwmertens%2Fsublime-nix%2Fblob%2Fmaster%2Fnix.YAML-tmLanguage&code=). Ignore the regex illegal character issues, the highlighter isn't 100% right since it does parse cleanly, as evidenced by http://nodeca.github.io/js-yaml/ which is what is linked from the yaml.org website.
